rrmngmnt/playbook_runner.py

```diff
@@ -158,6 +160,13 @@ def run(
                 for f in vars_files:
                     self.cmd.append("-e@{}".format(self._upload_file(f)))
 
+            if vault_password_file:
+                self.cmd.append(
+                    "--vault-password-file={}".format(
+                        self._upload_file(vault_password_file)
+                    )
+                )
+
```

Comment:
on line 188, I guess that you will have to upload that key file to host server, in case `upload_playbook=True`

Reply:
Well, you *always* have to pass local paths to both `vault_password_file` and to `vars_files`. No matter the value of `upload_playbook`. Let me update the docstring so this is clearer.

Reply:
`self.upload_file` returns path on remote host. and in case you pass `upload_playbook=False` you should pass `"--vault-password-file={}".format(vault_password_file)` , no?

Reply:
Well, you do have a point, but exactly the same also goes for `vars_files`. When I originally wrote this class and method, I never intended for user to upload any of those files manually. The parameters `upload_playbook` was implemented by @santos1709 and back then it was agreed that we wouldn't accommodate `vars_files` to it. However, it we want `vars_files` and `vault_password_file` to react on the value of `upload_playbook`, I think this requires bigger and more careful change. If that's the case, I suggest tracking it in a new issue/ticket.

Reply:
yeah, please open an issue. and we will merge this one.

Reply:
Tracked in #130 . I'll also create a ticket in our Jira board